### ISSUE TITLE
Unexpected errors are reported when type imported from metadata implements interface property in way that doesn't follow C# language rules

### ISSUE THREAD

**Issue by AlekseyTs** [dotnet/roslyn#33256]
```
        [ConditionalFact(typeof(CoreClrOnly))]
        public void ExplicitBase_134()
        {
            var ilSource = @"
.assembly extern System.Runtime
{
  .publickeytoken = (B0 3F 5F 7F 11 D5 0A 3A )                         // .?_....:
  .ver 4:2:1:0
}
.assembly extern System.Console
{
  .publickeytoken = (B0 3F 5F 7F 11 D5 0A 3A )                         // .?_....:
  .ver 4:1:1:0
}

.assembly ExplicitBase_134
{
  .hash algorithm 0x00008004
  .ver 0:0:0:0
}
.module ExplicitBase_134.dll
// MVID: {22493430-1D63-4B15-9B42-F868AA6D50D5}
.imagebase 0x10000000
.file alignment 0x00000200
.stackreserve 0x00100000
.subsystem 0x0003       // WINDOWS_CUI
.corflags 0x00000001    //  ILONLY
// Image base: 0x00AD0000


// =============== CLASS MEMBERS DECLARATION ===================

.class interface public abstract auto ansi B
       implements I1,
                  C,
                  I2
{
} // end of class B

.class interface public abstract auto ansi I1
       implements C
{
  .method public hidebysig specialname virtual final 
          instance char  C.get_F1() cil managed
  {
    .override C::get_F1
    // Code size       3 (0x3)
    .maxstack  8
    IL_0000:  ldc.i4.s   49
    IL_0002:  ret
  } // end of method I1::C.get_F1

  .property instance char C.F1()
  {
    .get instance char I1::C.get_F1()
  } // end of property I1::C.F1
} // end of class I1

.class interface public abstract auto ansi I2
       implements C
{
  .method public hidebysig specialname virtual final 
          instance void  C.set_F1(char 'value') cil managed
  {
    .override C::set_F1
    // Code size       12 (0xc)
    .maxstack  8
    IL_0000:  ldstr      ""set 2""
    IL_0005:  call       void [System.Console]System.Console::WriteLine(string)
    IL_000a:  nop
    IL_000b:  ret
  } // end of method I2::C.set_F1

  .property instance char C.F1()
  {
    .set instance void I2::C.set_F1(char)
  } // end of property I2::C.F1
} // end of class I2

.class interface public abstract auto ansi C
{
  .method public hidebysig newslot specialname abstract virtual 
          instance char  get_F1() cil managed
  {
  } // end of method C::get_F1

  .method public hidebysig newslot specialname abstract virtual 
          instance void  set_F1(char 'value') cil managed
  {
  } // end of method C::set_F1

  .property instance char F1()
  {
    .get instance char C::get_F1()
    .set instance void C::set_F1(char)
  } // end of property C::F1
} // end of class C

.class public auto ansi beforefieldinit D
       extends [System.Runtime]System.Object
       implements B,
                  I1,
                  C,
                  I2
{
  .method public hidebysig specialname rtspecialname 
          instance void  .ctor() cil managed
  {
    // Code size       8 (0x8)
    .maxstack  8
    IL_0000:  ldarg.0
    IL_0001:  call       instance void [System.Runtime]System.Object::.ctor()
    IL_0006:  nop
    IL_0007:  ret
  } // end of method A::.ctor
} // end of class A
";

            var source1 = @"
class A : B
{
    void Test()
    {
        _ = base(B).F1;
        base(B).F1 = 'x';
    }
}
";

            var reference = CompileIL(ilSource, prependDefaultHeader: false);

            var compilation1 = CreateCompilation(source1, references: new[] { reference },
                                                 options: TestOptions.DebugDll, targetFramework: TargetFramework.NetStandardLatest);
            compilation1.VerifyDiagnostics(
                // (2,11): error CS0535: 'A' does not implement interface member 'C.F1'
                // class A : B
                Diagnostic(ErrorCode.ERR_UnimplementedInterfaceMember, "B").WithArguments("A", "C.F1").WithLocation(2, 11)
                );

            var source2 = @"
class A : D, C
{
    char C.F1 { get => 'A'; set => System.Console.WriteLine(""set A"");}

    void Test()
    {
        System.Console.WriteLine(base(B).F1);
        base(B).F1 = 'x';
    }

    static void Main()
    {
        new A().Test();
        C d = new D();
        System.Console.WriteLine(d.F1);
        d.F1 = 'x';
    }
}
";
            var compilation2 = CreateCompilation(source2, references: new[] { reference },
                                                 options: TestOptions.DebugExe, targetFramework: TargetFramework.NetStandardLatest);
            compilation2.VerifyDiagnostics();

            var c = compilation2.GetTypeByMetadataName("C");
            var f1 = c.GetMember<PropertySymbol>("F1");
            var d = compilation2.GetTypeByMetadataName("D");

            Assert.Null(d.FindImplementationForInterfaceMember(f1));
            Assert.Equal("System.Char I1.F1.get", d.FindImplementationForInterfaceMember(f1.GetMethod).ToTestDisplayString());
            Assert.Equal("void I2.F1.set", d.FindImplementationForInterfaceMember(f1.SetMethod).ToTestDisplayString());

            CompileAndVerify(compilation2, expectedOutput:
@"
1
set 2
1
set 2
"
);

            var source3 = @"
class A : D, B
{
    static void Main()
    {
        C a = new A();
        System.Console.WriteLine(a.F1);
        a.F1 = 'x';
    }
}
";
            var compilation3 = CreateCompilation(source3, references: new[] { reference },
                                                 options: TestOptions.DebugExe, targetFramework: TargetFramework.NetStandardLatest);
            compilation3.VerifyDiagnostics();
            CompileAndVerify(compilation3, expectedOutput:
@"
1
set 2
"
);
        }
```

Observed: An error is reported for compilation1.
Expected: No errors should be reported, compilation2 and compilation3 demonstrate that runtime considers Interface C fully implemented by interface B.

**Comment by AlekseyTs** [comment]
Here is one more variation:
```
        [ConditionalFact(typeof(CoreClrOnly))]
        public void ExplicitBase_135()
        {
            var ilSource = @"
.assembly extern System.Runtime
{
  .publickeytoken = (B0 3F 5F 7F 11 D5 0A 3A )                         // .?_....:
  .ver 4:2:1:0
}
.assembly extern System.Console
{
  .publickeytoken = (B0 3F 5F 7F 11 D5 0A 3A )                         // .?_....:
  .ver 4:1:1:0
}

.assembly ExplicitBase_135
{
  .hash algorithm 0x00008004
  .ver 0:0:0:0
}
.module ExplicitBase_135.dll
// MVID: {22493430-1D63-4B15-9B42-F868AA6D50D5}
.imagebase 0x10000000
.file alignment 0x00000200
.stackreserve 0x00100000
.subsystem 0x0003       // WINDOWS_CUI
.corflags 0x00000001    //  ILONLY
// Image base: 0x00AD0000


// =============== CLASS MEMBERS DECLARATION ===================

.class interface public abstract auto ansi B
       implements I1,
                  C,
                  I2
{
} // end of class B

.class interface public abstract auto ansi I1
       implements C
{
  .method public hidebysig specialname virtual final 
          instance char  C.get_F1() cil managed
  {
    .override C::get_F1
    // Code size       3 (0x3)
    .maxstack  8
    IL_0000:  ldc.i4.s   49
    IL_0002:  ret
  } // end of method I1::C.get_F1
} // end of class I1

.class interface public abstract auto ansi I2
       implements C
{
  .method public hidebysig specialname virtual final 
          instance void  C.set_F1(char 'value') cil managed
  {
    .override C::set_F1
    // Code size       12 (0xc)
    .maxstack  8
    IL_0000:  ldstr      ""set 2""
    IL_0005:  call       void [System.Console]System.Console::WriteLine(string)
    IL_000a:  nop
    IL_000b:  ret
  } // end of method I2::C.set_F1
} // end of class I2

.class interface public abstract auto ansi C
{
  .method public hidebysig newslot specialname abstract virtual 
          instance char  get_F1() cil managed
  {
  } // end of method C::get_F1

  .method public hidebysig newslot specialname abstract virtual 
          instance void  set_F1(char 'value') cil managed
  {
  } // end of method C::set_F1

  .property instance char F1()
  {
    .get instance char C::get_F1()
    .set instance void C::set_F1(char)
  } // end of property C::F1
} // end of class C

.class public auto ansi beforefieldinit D
       extends [System.Runtime]System.Object
       implements B,
                  I1,
                  C,
                  I2
{
  .method public hidebysig specialname rtspecialname 
          instance void  .ctor() cil managed
  {
    // Code size       8 (0x8)
    .maxstack  8
    IL_0000:  ldarg.0
    IL_0001:  call       instance void [System.Runtime]System.Object::.ctor()
    IL_0006:  nop
    IL_0007:  ret
  } // end of method A::.ctor
} // end of class A
";

            var source1 = @"
class A : B
{
    void Test()
    {
        _ = base(B).F1;
        base(B).F1 = 'x';
    }
}
";
            var reference = CompileIL(ilSource, prependDefaultHeader: false);

            var compilation1 = CreateCompilation(source1, references: new[] { reference },
                                                    options: TestOptions.DebugDll, targetFramework: TargetFramework.NetStandardLatest);
            compilation1.VerifyDiagnostics(
                // (2,11): error CS0535: 'A' does not implement interface member 'C.F1'
                // class A : B
                Diagnostic(ErrorCode.ERR_UnimplementedInterfaceMember, "B").WithArguments("A", "C.F1").WithLocation(2, 11)
                );

            var source2 = @"
class A : D, C
{
    char C.F1 { get => 'A'; set => System.Console.WriteLine(""set A"");}

    void Test()
    {
        System.Console.WriteLine(base(B).F1);
        base(B).F1 = 'x';
    }

    static void Main()
    {
        new A().Test();
        C d = new D();
        System.Console.WriteLine(d.F1);
        d.F1 = 'x';
    }
}
";
            var compilation2 = CreateCompilation(source2, references: new[] { reference },
                                                 options: TestOptions.DebugExe, targetFramework: TargetFramework.NetStandardLatest);
            compilation2.VerifyDiagnostics();

            var c = compilation2.GetTypeByMetadataName("C");
            var f1 = c.GetMember<PropertySymbol>("F1");
            var d = compilation2.GetTypeByMetadataName("D");

            Assert.Null(d.FindImplementationForInterfaceMember(f1));
            Assert.Equal("System.Char I1.C.get_F1()", d.FindImplementationForInterfaceMember(f1.GetMethod).ToTestDisplayString());
            Assert.Equal("void I2.C.set_F1(System.Char value)", d.FindImplementationForInterfaceMember(f1.SetMethod).ToTestDisplayString());

            CompileAndVerify(compilation2, expectedOutput:
@"
1
set 2
1
set 2
"
);

            var source3 = @"
class A : D, B
{
    static void Main()
    {
        C a = new A();
        System.Console.WriteLine(a.F1);
        a.F1 = 'x';
    }
}
";
            var compilation3 = CreateCompilation(source3, references: new[] { reference },
                                                 options: TestOptions.DebugExe, targetFramework: TargetFramework.NetStandardLatest);
            compilation3.VerifyDiagnostics();
            CompileAndVerify(compilation3, expectedOutput:
@"
1
set 2
"
);
        }
```

**Comment by AlekseyTs** [comment]
Here is a scenario with an event:
```

        [ConditionalFact(typeof(CoreClrOnly))]
        [WorkItem(33256, "https://github.com/dotnet/roslyn/issues/33256")]
        public void ExplicitBase_146()
        {
            var ilSource = @"
.assembly extern System.Runtime
{
  .publickeytoken = (B0 3F 5F 7F 11 D5 0A 3A )                         // .?_....:
  .ver 4:2:1:0
}
.assembly extern System.Console
{
  .publickeytoken = (B0 3F 5F 7F 11 D5 0A 3A )                         // .?_....:
  .ver 4:1:1:0
}

.assembly ExplicitBase_146
{
  .hash algorithm 0x00008004
  .ver 0:0:0:0
}
.module ExplicitBase_146.dll
// MVID: {22493430-1D63-4B15-9B42-F868AA6D50D5}
.imagebase 0x10000000
.file alignment 0x00000200
.stackreserve 0x00100000
.subsystem 0x0003       // WINDOWS_CUI
.corflags 0x00000001    //  ILONLY
// Image base: 0x00AD0000


// =============== CLASS MEMBERS DECLARATION ===================

.class interface public abstract auto ansi B
       implements I1,
                  C,
                  I2
{
} // end of class B

.class interface public abstract auto ansi I1
       implements C
{
  .method public hidebysig specialname virtual final 
          instance void  C.add_F1(class [System.Runtime]System.Action 'value') cil managed
  {
    .override C::add_F1
    // Code size       12 (0xc)
    .maxstack  8
    IL_0000:  ldstr      ""add 1""
    IL_0005:  call       void [System.Console]System.Console::WriteLine(string)
    IL_000a:  nop
    IL_000b:  ret
  } // end of method I1::C.add_F1
} // end of class I1

.class interface public abstract auto ansi I2
       implements C
{
  .method public hidebysig specialname virtual final 
          instance void  C.remove_F1(class [System.Runtime]System.Action 'value') cil managed
  {
    .override C::remove_F1
    // Code size       12 (0xc)
    .maxstack  8
    IL_0000:  ldstr      ""remove 2""
    IL_0005:  call       void [System.Console]System.Console::WriteLine(string)
    IL_000a:  nop
    IL_000b:  ret
  } // end of method I2::C.remove_F1
} // end of class I2

.class interface public abstract auto ansi C
{
  .method public hidebysig newslot specialname abstract virtual 
          instance void  add_F1(class [System.Runtime]System.Action 'value') cil managed
  {
  } // end of method C::add_F1

  .method public hidebysig newslot specialname abstract virtual 
          instance void  remove_F1(class [System.Runtime]System.Action 'value') cil managed
  {
  } // end of method C::remove_F1

  .event [System.Runtime]System.Action F1
  {
    .addon instance void C::add_F1(class [System.Runtime]System.Action)
    .removeon instance void C::remove_F1(class [System.Runtime]System.Action)
  } // end of event C::F1
} // end of class C

.class public auto ansi beforefieldinit D
       extends [System.Runtime]System.Object
       implements B,
                  I1,
                  C,
                  I2
{
  .method public hidebysig specialname rtspecialname 
          instance void  .ctor() cil managed
  {
    // Code size       8 (0x8)
    .maxstack  8
    IL_0000:  ldarg.0
    IL_0001:  call       instance void [System.Runtime]System.Object::.ctor()
    IL_0006:  nop
    IL_0007:  ret
  } // end of method A::.ctor
} // end of class A
";

            var source1 = @"
class A : B
{
    void Test()
    {
        base(B).F1 += null;
        base(B).F1 -= null;
    }
}
";
            var reference = CompileIL(ilSource, prependDefaultHeader: false);

            var compilation1 = CreateCompilation(source1, references: new[] { reference },
                                                    options: TestOptions.DebugDll, targetFramework: TargetFramework.NetStandardLatest);
            compilation1.VerifyDiagnostics(
                // (2,11): error CS0535: 'A' does not implement interface member 'C.F1'
                // class A : B
                Diagnostic(ErrorCode.ERR_UnimplementedInterfaceMember, "B").WithArguments("A", "C.F1").WithLocation(2, 11)
                );

            var source2 = @"
class A : D, C
{
    event System.Action C.F1 { add => System.Console.WriteLine(""add A""); remove => System.Console.WriteLine(""remove A"");}

    void Test()
    {
        base(B).F1 += null;
        base(B).F1 -= null;
    }

    static void Main()
    {
        new A().Test();
        C d = new D();
        d.F1 += null;
        d.F1 -= null;
    }
}
";
            var compilation2 = CreateCompilation(source2, references: new[] { reference },
                                                 options: TestOptions.DebugExe, targetFramework: TargetFramework.NetStandardLatest);
            compilation2.VerifyDiagnostics();

            var c = compilation2.GetTypeByMetadataName("C");
            var f1 = c.GetMember<EventSymbol>("F1");
            var d = compilation2.GetTypeByMetadataName("D");

            Assert.Null(d.FindImplementationForInterfaceMember(f1));
            Assert.Equal("void I1.C.add_F1(System.Action value)", d.FindImplementationForInterfaceMember(f1.AddMethod).ToTestDisplayString());
            Assert.Equal("void I2.C.remove_F1(System.Action value)", d.FindImplementationForInterfaceMember(f1.RemoveMethod).ToTestDisplayString());

            CompileAndVerify(compilation2, expectedOutput:
@"
add 1
remove 2
add 1
remove 2
"
);

            var source3 = @"
class A : D, B
{
    static void Main()
    {
        C a = new A();
        a.F1 += null;
        a.F1 -= null;
    }
}
";
            var compilation3 = CreateCompilation(source3, references: new[] { reference },
                                                 options: TestOptions.DebugExe, targetFramework: TargetFramework.NetStandardLatest);
            compilation3.VerifyDiagnostics();
            CompileAndVerify(compilation3, expectedOutput:
@"
add 1
remove 2
"
);
        }
```

**Comment by AlekseyTs** [comment]
The error is probably fine, C# language can have stricter requirements, but I think it makes sense to get confirmation from LDM and see this should have any implications for base-access (see ```compilation2```).

**Comment by gafter** [comment]
I'm not sure I understand the issue.  The C# compiler is giving an error when an abstract interface member is not implemented (according to the C# rules).  So the C# compiler is following the C# language specification.  Are you saying that's a bug?  I understand we can imagine a C# compiler following looser rules than required by the C# language specification, but I can't tell if that's what you think the compiler should be doing.

**Comment by gafter** [comment]
@AlekseyTs Can you please clarify if you are reporting a bug (compiler not behaving to spec) or something else?
